admin/institutions/views.py

```diff
@@ -173,28 +174,62 @@ class DeleteInstitution(PermissionRequiredMixin, DeleteView):
     success_url = reverse_lazy('institutions:list')
 
     def delete(self, request, *args, **kwargs):
-        institution = Institution.objects.get(id=self.kwargs['institution_id'])
+        institution = Institution.objects.get_all_institutions().get(id=self.kwargs['institution_id'])
         if institution.nodes.count() > 0:
             return redirect('institutions:cannot_delete', institution_id=institution.pk)
         return super(DeleteInstitution, self).delete(request, *args, **kwargs)
 
     def get(self, request, *args, **kwargs):
-        institution = Institution.objects.get(id=self.kwargs['institution_id'])
+        institution = Institution.objects.get_all_institutions().get(id=self.kwargs['institution_id'])
         if institution.nodes.count() > 0:
             return redirect('institutions:cannot_delete', institution_id=institution.pk)
         return super(DeleteInstitution, self).get(request, *args, **kwargs)
 
     def get_object(self, queryset=None):
-        institution = Institution.objects.get(id=self.kwargs['institution_id'])
+        institution = Institution.objects.get_all_institutions().get(id=self.kwargs['institution_id'])
         return institution
 
 
+class DeactivateInstitution(PermissionRequiredMixin, UpdateView):
+    template_name = 'institutions/confirm_deactivate.html'
+    permission_required = 'osf.change_institution'
+    raise_exception = True
+    model = Institution
+    form_class = InstitutionForm
+
+    def get_object(self, queryset=None):
+        return Institution.objects.get_all_institutions().get(id=self.kwargs.get('institution_id'))
+
+    def post(self, request, *args, **kwargs):
+        institution = self.get_object()
+        institution.deactivated = timezone.now()
+        institution.save()
```

Comment:
As discussed, `.save()` calls `update_search()` before calling `super().save()` ([here](https://github.com/CenterForOpenScience/osf.io/blob/91a7ca387d9fd5d56d564df23e0f26b026342498/osf/models/institution.py#L150-L152)). This leads to the search engine update seeing the outdated affiliations.

The fix is to call `super().save()` first. This has an unpleasant side effect that `dirty_fields` will no longer be available. To solve this, update search only after save if there are dirty fields before save.

We may need to update/fix existing unit tests since 1) `update_search()` will be taken out of `save()` and 2) there is no need to have a customized `save()` function for the institution model.

Reply:
Opted to modify the behavior directly in `institution.save()` since we do want to continue updating search for institutions in all places where they are saved.